src/components/templates/PosterHall/PosterHall.tsx

```diff
@@ -12,13 +12,35 @@ import { Button } from "components/atoms/Button";
 import { PosterPreview } from "./components/PosterPreview";
 import { PosterHallSearch } from "./components/PosterHallSearch";
 
+import { useUser } from "hooks/useUser";
+import { MyPersonalizedPosters } from "types/User";
+import { PosterPageVenue, PersonalizedPoster } from "types/venues";
+import { isTruthy } from "utils/types";
+
 import "./PosterHall.scss";
 
+const prepareForMyPosters = (usersPosters: MyPersonalizedPosters) => (
+  poster: WithId<PosterPageVenue>
+): WithId<PersonalizedPoster> => {
+  return {
+    ...poster,
+    isSaved: isTruthy(
+      poster.id && usersPosters[poster.id]?.includes(poster.id)
+    ),
+  };
+};
```

Comment:
This should be in a `utils` file of some sort probably (if it's even still needed?)

Reply:
no longer relevent